tests/integration/MediaWiki/ExternalLinksTest.php

```diff
@@ -70,6 +70,22 @@ public function testPageIdsContainingALink() {
 		$this->assertSame( [ '1' ], $result );
 	}
 
+	public function testPageIdsContainingLinksWithDifferentUserinfo(): void {
+		$userinfoAndPageIds = [
+			[ 'me:mypassword', 1 ],
+			[ 'someoneelse:theirpassword', 2 ],
+		];
+		foreach ( $userinfoAndPageIds as [ $userinfo, $pageId ] ) {
+			$rows = $this->getExternalLinksRows( "http://$userinfo@example.com/", $pageId );
+			$this->db->insert( 'externallinks', $rows, __METHOD__ );
+		}
+
+		$externalLinks = $this->newExternalLinks();
+		$result = $externalLinks->pageIdsContainingUrl( 'http://me:mypassword@example.com/' );
```

Comment:
Should we change name on this now as the function now requires an exact match?

```
pageIdsMatchingUrl
```

maybe?

Reply:
I think it was always meant to be an exact match – “containing” refers to the page having more content than just the one URL (and a page can contain multiple URLs).